source/common/network/connection_impl.h

```diff
@@ -74,7 +74,7 @@ class ConnectionImpl : public virtual Connection,
   uint32_t readBufferLimit() const override { return read_buffer_limit_; }
 
   // Network::BufferSource
-  Buffer::Instance& getReadBuffer() override { return read_buffer_; }
+  Buffer::Instance& getReadBuffer() override { return *read_buffer_; }
   Buffer::Instance& getWriteBuffer() override { return *current_write_buffer_; }
```

Comment:
There's both a write_buffer_ and a current_write_buffer_. write_buffer_ is initialized via factory, but current_write_buffer_ is not (in fact it looks to me like it's always nullptr, so this line seems especially odd). This isn't a line changed in this PR, but it would be good to clarify...

@mattklein123 any idea what current_write_buffer_ was supposed to be for?